### DISCUSSION TITLE
SQLAlchemy + Asyncpg support

### DISCUSSION THREAD

**Discussion by bittermandel** [fastapi-users/fastapi-users#717] [Show and tell]
Hey!

I forked and reworked the sqlalchemy backend with support for AsyncPG + sessions. This was to get rid of the dependency on encode/databases and work purely with upstream SQLAlchemy.

If it's of any interest I'm more than happy to transfer this to the fastapi-users organization.

You can find the fork here: https://github.com/BPL-Development-Team/fastapi-users-db-sqlalchemy-asyncpg

**Comment by frankie567** [comment]
Hi!

That's indeed interesting, thanks! However, why a focus on `asyncpg`? SQLAlchemy can work with several DBMS so I think it may work with SQLite or MySQL likewise. I may take inspiration from it, if you don't mind :)

As a side note, I'm also working on an adapter for SQLModel (#698) ; which could also be interesting.

Cheers!
